src/Cake.Core/IO/ProcessRunner.cs

```diff
@@ -52,6 +55,20 @@ public IProcess Start(FilePath filePath, ProcessSettings settings)
             // Get the arguments.
             var arguments = settings.Arguments ?? new ProcessArgumentBuilder();
 
+#if NETCORE
+            if (_environment.Platform.IsUnix() &&
+                fileName.EndsWith(".exe", StringComparison.OrdinalIgnoreCase) &&
+                _fileSystem.GetFile(fileName).IsCLRAssembly())
+            {
+                if (!settings.Silent)
+                {
+                    _log.Verbose(Verbosity.Diagnostic, "{0} is a .NET Framework executable, will try execute using Mono.", fileName);
```

Comment:
If `mono` is not installed on the machine that this is running on, this is obviously going to fail.  Do we need to provide a sensible error message if/when this happens?  Should we first attempt to check to see if mono is installed prior to doing this?

Do we need to provide a configuration value to prevent this "magic" from happening?  i.e. is there a scenario where when running on .NET Core, on Unix with a CLRAssembly, that the end user doesn't want to run with mono?

Reply:
Basically if ends with `.exe` and has all the identifiers a clr assembly has.

.NET Core has the `.dll` extension, and native shims are extension less on unix (and don't have the needed PE headers)

Could probably bring in the tool resolver to try find mono, warn and not execute it if not found.

Only thing I could think of is WSL which can execute windows binaries from the linux side. 
But that feels like a real non best practice and will require the tool to be installed in path on windows side.

Reply:
@devlead what were your thoughts around a configuration value to control whether this "magic" happens or not?

Reply:
Sure we could add an opt out for this, any suggestions around naming?

Reply:
Don't have good imagination today;) So unless anyone comes up with a better name opt-out setting will be `Settings_SkipFullFrameworkMonoFallbackOnUnixCoreClr`.

Reply:
A suggestion from @patriksvensson which I think makes sense is `Settings_NoMonoCoersion` and more detail about exactly what it is doing can be added in the documentation.